pkg/kubectl/resource_filter.go

```diff
@@ -58,8 +58,18 @@ func filterPods(obj runtime.Object, options PrintOptions) bool {
 
 // Filter loops through a collection of FilterFuncs until it finds one that can filter the given resource
 func (f Filters) Filter(obj runtime.Object, opts *PrintOptions) (bool, error) {
-	// check if the object is unstructured. If so, let's attempt to convert it to a type we can understand
-	// before apply filter func.
+	for _, filter := range f {
```

Comment:
Suggest calling `DecodeUnknownObject` again inside `Filter`, like what you did in `PrintObj`, just in case the caller passes an Unknown/Unstructured type obj